### PR TITLE
Include aggregations and suggest in SearchTemplateResponse

### PR DESCRIPTION
**Pull request by eNorby1**
### Description
Added support for aggregations and suggest in SearchTemplateResponse

### Issues Resolved
Resolves [#456](https://github.com/opensearch-project/opensearch-java/issues/456)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — CHANGES_REQUESTED** [review]
This is great! Just needs tests, please.

**Comment by eNorby1** [comment]
Hi @dblock and others, can you please check? :)

**Review by dblock — APPROVED** [review]
This looks great! Super idea. I don't see any downsides, @reta?

**Comment by reta** [comment]
> This looks great! Super idea. I don't see any downsides, @reta?

@eNorby1 I really like the direction you ended up at, I would suggest minor design change: introduce intermediary `SearchResult<TDocument>` class so both `SearchResponse` and `SearchTemplateResponse` could inherit from it.

Why I am suggestion that instead of subclassing `SearchResponse` directly: besides the fact that  `SearchTemplateResponse`  hides some static fields of the `SearchResponse` (not a big deal but could be surprising), the change gives an impression that`SearchTemplateResponse` is not really needed and  `SearchResponse` could be used instead (with intermediary class we basically inherit implementation only).

**Comment by eNorby1** [comment]
@reta makes sense. I'll have a look, thanks

**Comment by eNorby1** [comment]
Hi @reta I did the change you suggested, can you please check?

File: java-client/src/main/java/org/opensearch/client/opensearch/core/search/SearchResult.java [hunk 1]
```diff
@@ -0,0 +1,639 @@
+/*
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/opensearch/core/search/SearchResult.java:1]: Super nit, but we could only keep this part of the header, this is new file and is not inherited from Elasticsearch

**Comment by reta** [comment]
> Hi @reta I did the change you suggested, can you please check?

Thanks a lot @eNorby1 , it looks great, nit comment regarding the copyright header, LGTM otherwise!

File: java-client/src/main/java/org/opensearch/client/opensearch/core/search/SearchResult.java [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+//typedef: _global.search.SearchResult
+
+public abstract class SearchResult<TDocument> implements JsonpSerializable {
```
[the remark this answers is not in the document]
Comment (reta) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/search/SearchResult.java:64]: Fair enough, thank you

**Review by reta — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**